src/Pipes/GZip.hs

```diff
@@ -62,8 +62,8 @@ decompress' = Pipes.Zlib.decompress' gzWindowBits
 compress
   :: MonadIO m
   => Pipes.Zlib.CompressionLevel
-  -> Proxy x' x () B.ByteString m r -- ^ Decompressed stream
-  -> Proxy x' x () B.ByteString m r -- ^ Compressed stream
+  -> Producer B.ByteString m r -- ^ Decompressed stream
+  -> Producer' B.ByteString m r -- ^ Compressed stream
```

Comment:
@dimbleby this actually became less generic here, since it's restricting the previous passed in`x'` to `Pipes.X`, and`x` to `()`. Yes, the resulting `Producer'` is equally polymorphic, but not the passed in one. Do your changes actually mandate this restriction?